src/resolvers/taxonomyResolvers.ts

```diff
@@ -0,0 +1,180 @@
+  fetchLearningpath,
+  fetchNode,
+  fetchNodeByContentUri,
+  queryNodes,
+} from "../api";
+import {
+  GQLLearningpath,
+  GQLMeta,
+  GQLNode,
+  GQLNodeChildrenArgs,
+  GQLQueryNodeArgs,
+  GQLQueryNodeByArticleIdArgs,
+  GQLQueryNodesArgs,
+  GQLTaxonomyEntity,
+} from "../types/schema";
+import { nodeToTaxonomyEntity } from "../utils/apiHelpers";
+import {
+  articleToMeta,
+  filterMissingArticles,
+  getArticleIdFromUrn,
+  getLearningpathIdFromUrn,
+} from "../utils/articleHelpers";
+
+const pickContextValues = (node: Node, language: string, rootId?: string, parentId?: string) => {
+  const visibleContexts = node.contexts.filter((c) => c.isVisible && !c.rootId.startsWith("urn:programme"));
+  // A resource can have several context in one subject. rootId and parentId helps picking the right one.
+  const rootContexts = rootId ? visibleContexts.filter((c) => c.rootId === rootId) : visibleContexts;
+  const parentContexts = parentId ? rootContexts.filter((c) => c.parentIds.includes(parentId)) : rootContexts;
+
+  const selectedCtx = parentContexts?.[0];
+  const path = selectedCtx?.path || node.path;
+  const rank = selectedCtx?.rank;
+  const contextId = selectedCtx?.contextId;
+  const relevanceId = selectedCtx?.relevanceId;
+  const entity = nodeToTaxonomyEntity({ ...node, contexts: visibleContexts }, language);
+  return { ...entity, contextId, path, rank, relevanceId };
+};
+
+export const Query = {
+  async node(
+    _: any,
+    { id, rootId, parentId, contextId }: GQLQueryNodeArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode | undefined> {
+    if (id) {
+      const node = await context.loaders.nodeLoader.load({ id, rootId, parentId });
+      return pickContextValues(node, context.language, rootId, parentId);
+    }
+    if (contextId) {
+      const nodes = await queryNodes({ contextId, includeContexts: true, filterProgrammes: true }, context);
+      if (nodes.length === 0) {
+        throw new Error(`No node found with contextId: ${contextId}`);
+      }
+      const node = nodes[0];
+      return node ? nodeToTaxonomyEntity(node, context.language, contextId) : undefined;
+    }
+    throw new Error("Missing id or contextId");
+  },
+  async nodes(
+    _: any,
+    { nodeType, contentUri, filterVisible, metadataFilterKey, metadataFilterValue, ids }: GQLQueryNodesArgs,
+    context: ContextWithLoaders,
+  ): Promise<GQLNode[]> {
+    const params = {
+      ids: ids,
+      includeContexts: true,
+      key: metadataFilterKey,
+      value: metadataFilterValue,
+      filterProgrammes: true,
+      language: context.language,
+    };
+    let nodes: Node[] = [];
+    if (contentUri) {
+      nodes = await queryNodes({ contentURI: contentUri, ...params }, context);
+    } else if (nodeType) {
+      nodes = await queryNodes({ nodeType: nodeType, ...params }, context);
+    }
+    const filtered = filterVisible ? nodes.filter((node) => node.contexts.find((context) => context.isVisible)) : nodes;
```

Comment:
Hvorfor returnerer vi egentlig skjulte kontekster fra tax? Hvilke tilfeller er det vi har bruk for de?

Reply:
Kan være nyttig i ed. Ser her at eg kan sende inn parameter til tax for å unngå denne filtreringa.

Reply:
Burde isåfall kanskje defaulte til at skjulte kontekster ikke sendes ut?